<h1><b><center></center>Exercícios Cap1 Matemática Financeira - Samanez</b></h1>

In [103]:
from sympy import *
#Importando a biblioteca sympy para simplificar contas

<p> 3. Calcular o rendimento de um capital de &#36;80.000 aplicado durante 28 dias à taxa de juros simples
de 26% a.m</p>



In [104]:
taxa_dias = (26/30)/100 #Taxa ao dia
rendimento = 80000 * taxa_dias * 28 #calculo de juros simples

f'R$ {rendimento:,.2f}' #formata o valor em real

'R$ 19,413.33'

<p> 6. Um capital aplicado transformou-se em $13.000. Considerando uma taxa de juros simples de 42% a.a.
e uma remuneração de &#36;4.065,29, determinar o prazo da aplicação. </p>



In [105]:
n = symbols('n') #define o simbolo

taxa_meses = (42/100)/12 #taxa em meses

eq_rendimentos = Eq((13000*taxa_meses*n)/(1+(taxa_meses*n)), 4065.29) #equacao J = (s*i*n)/1 + i*n

valor_n = solve(eq_rendimentos, n)[0] #usa a formula eq_rendimentos para resolver o valor de n

round(valor_n) #round para aproximar valor ao inteiro mais proximo

13

<p> 9. Um capital acrescido de seus juros de 21 meses soma $156.400. O mesmo capital diminuído de seus
juros de nove meses é reduzido a &#36;88.400. Calcular o capital e a taxa de juros simples obtidos. </p>



In [106]:
i, p = symbols('i, p') #define simbolos

eq1 = Eq(p * (1 + 21 * i), 156400) #montante1
eq2 = Eq(p * (1 - 9 * i), 88400) #montante2

divisao = (p * (1 + 21 * i)) / (p * (1 - 9 * i)) #divide equações montante1 e 2

divisao_simplificada = Eq(divisao, 156400 / 88400) #equacao i sem p

taxa_juros = solve(divisao_simplificada, i)[0] #esta em forma decimal e em mensal

eq3 = Eq(p*(1+taxa_juros*21), 156400)

aplicacao = solve(eq3)[0]

f"{taxa_juros*100*12:.0f}% a.m.", f'R$ {aplicacao:,.2f}' #formato da taxa em percentual a.m., formato do capital em real

('25% a.m.', 'R$ 108,800.00')

<p> 12. Um capital foi aplicado a juros simples de 42% a.a. durante 50 dias. Calcular o capital, sabendo-se
que, se a diferença entre ele e os juros obtidos fosse aplicada à mesma taxa, renderia &#36;988,75 em
um trimestre. </p>



In [107]:
p = symbols('p')
taxa = (42/100)/360

eq = Eq((p-(p*taxa*50))*90*taxa, 988.75)

aplicacao = solve(eq)[0]

f'R$ {aplicacao:,.2f}' #formata o valor em real

'R$ 10,000.00'

<p> 15. Há 13 meses e dez dias, um capital de &#36;10.000 foi aplicado à taxa de juros simples de 6% a.a.
Se hoje fosse aplicada a importância de $8.000 a juros simples de 12% a.a., e o primeiro capital
continuar aplicado à mesma taxa, em que prazo os montantes respectivos seriam iguais?
 </p>



In [108]:
tempo_dias = (13*30)+10 #transforma os messes em dias
taxa1 = (6/100)/360 #taxa do capital de 10000 em dias
taxa2 = (12/100)/360 #taxa do capital de 8000 em dias
n, j = symbols('n, j') #define simbolos

eq2 = 8000*(1 + taxa2 * n) #equacao montante para a aplicacao de 8000

eq1 =10000*(1 + taxa1 * (n+400)) #equacao montante para a aplicacao de 10000, n + tempo_dias pois ele ja estava aplicado a 400 dias

eq1_eq2 = Eq(eq1, eq2) #iguala a eq1 e eq2

valor_dias = solve(eq1_eq2, n) #resolve a eq1 e eq2 para o valor de n

valor_dias[0]/360 #divido por 360 para valor em anos

7.40740740740741

<p> 21. Dois capitais foram colocados a juros simples, o primeiro à taxa de 20% a.a. e o segundo à de 40% a.a. Calcular os capitais, sabendo-se que, somados, eles montam &#36;500 e que os dois, em um ano, renderam juros totais de $130.
 </p>



In [109]:
p1, p2 = symbols('p1 p2')
capitais = Eq(p1 + p2, 500) #Dividindo a soma pela taxa 0,2
valor_p1 = solve(capitais, p1) # resolve a soma dos capitais para p1

somando_juros = Eq((valor_p1[0] * 0.2 * 1 + p2 * 0.4 * 1), 130) #usa o valor de p1 e coloca na soma dos juros
valor_p2 = solve(somando_juros, p2) #resolve a soma dos juros para o valor de p2
valor_p1 = 500 - valor_p2[0] #valor da soma dos capitais menos o valor do capital 2 (p2) e atualiza o valor_p1

f'R$ {valor_p1:,.2f}', f'R$ {valor_p2[0]:,.2f}' #formata o valores em real

('R$ 350.00', 'R$ 150.00')

<p> 27. Determinar: <br>
a. a taxa de juros simples anual que produz um rendimento de &#36;60 em 36 dias a partir de um
capital de &#36;2.000; <br>
b. a taxa de juros simples mensal que produz um rendimento de &#36;6.000 em 30 meses a partir
de um capital de &#36;8.000; <br>
c. a taxa de juros simples anual embutida na compra de um bem cujo valor à vista é de &#36;3.000,
pelo qual será paga uma entrada de $1.000 mais um pagamento de &#36;2.200 para 60 dias.
 </p>



In [110]:
i = symbols('i') #define simbolos

#resolucao a
eq_a = Eq(36*2000*i, 60) #equacao dos juros obtidos

valor_ai = solve(eq_a, i)[0] #resolve a equacao para i

jurosa_anual = (valor_ai*100)*360 #transforma os juros diarios em anual

#resolucao b
eq_b = Eq(30*8000*i, 6000) #equacao dos juros obtidos

valor_bi = solve(eq_b, i)[0] #resolve a equacao para i

jurosb_mensal = (valor_bi*100) #juros mensal em percentual

#resolucao c
#duas formas de fazer, com a formula de S, ou calculando os juros de 2000 a 60dias, farei segundo modo
eq_c = Eq(2000*60*i, 200) #equacao dos juros obtidos

valor_ci = solve(eq_c, i)[0]
juros_canual = valor_ci*100*360 #juros percentual anual

f"{jurosa_anual:.0f}% a.a.", f"{jurosb_mensal.evalf(2):.1f}% a.m.", f"{juros_canual:.0f}% a.a."
#.evalf(2) para converter resultado da fraçao em numero decimal
#formatacao para saida em numero percentual

('30% a.a.', '2.5% a.m.', '60% a.a.')

<p> 30. No dia 26 de maio foi contratado um empréstimo de &#36;7.000 a juros simples de 24% a.a. para ser
totalmente liquidado em 90 dias. No dia 16 de junho foram amortizados &#36;3.000, e no dia 11 de
julho, &#36;2.500. Determinar a data de vencimento da dívida e o valor da quantia que deverá ser paga
naquela data para liquidar a dívida. (Data focal: 90&deg;
 dia.)
 </p>



In [111]:
#facilmente podemos descobrir a data de vencimento olhando a tabela de contagem de dias
# Data de vencimento 24 de agosto, tambem usando a tabela vemos a diferenca entre os dias
#calculo para o valor da quantia
taxa_decimal = 24/100
valor_quantia = ( 7000*(1+(taxa_decimal/360)*90)) - ( 3000*(1+(taxa_decimal/360)*69) ) - ( 2500*(1+(taxa_decimal/360)*44) )

f'R$ {valor_quantia:,.2f}' #formata valor de saida em real

'R$ 1,708.67'

<p> 33. Uma aplicação financeira foi iniciada no dia 2 de junho com &#36;2.000. Posteriormente, foram
efetuados dois depósitos adicionais de &#36;500 e de &#36;300 nos dias 8 e 16 e um saque de &dollar;200 no dia
26 de junho. Se inicialmente foi contratada uma taxa de juros simples de 28% a.a., que depois
baixou para 26% a.a. no dia 16 de junho, calcular o saldo disponível em 1&deg;
 de julho.
 </p>



In [117]:
#usando a tabua de contagem de dias entre datas descobrimos a diferenca entre os dias.
taxa_dias1 = (28/100)/360 #taxa inicial em dias e decimal
taxa_dias2 = (26/100)/360 #taxa final em dias e decimal

saldo_disponivel = 2000*(1+taxa_dias1*29)+500*(1+taxa_dias1*23)+300*(1+taxa_dias2*15)-200*(1+taxa_dias2*5)

f'R$ {saldo_disponivel:,.2f}' #formata valor em real

'R$ 2,656.58'